includes/class-wc-stripe-blocks-support.php

```diff
@@ -413,10 +415,38 @@ private function add_order_meta( \WC_Order $order, $payment_request_type ) {
 	 */
 	public function get_supported_features() {
 		$gateways = WC()->payment_gateways->get_available_payment_gateways();
+
 		if ( isset( $gateways['stripe'] ) ) {
 			$gateway = $gateways['stripe'];
-			return array_filter( $gateway->supports, [ $gateway, 'supports' ] );
+		} elseif ( $this->is_upe_method_available( $gateways ) ) {
+			$gateway = WC_Stripe::get_instance()->get_main_stripe_gateway();
+		} else {
+			return [];
+		}
+
+		return array_filter( $gateway->supports, [ $gateway, 'supports' ] );
+	}
```

Comment:
For some background, this change and the one above are essentially the same underlying issue. The blocks support file was reliant on the 'stripe' payment method being in the list of available payment methods. Because the "stripe" payment method is the main UPE gateway and is only available if cards are enabled, disabling cards would prevent these two blocks supportive functions to no longer work as expected. 

I've changed them both so if Stripe isn't available, it will check if any of our UPE methods are available. If at least 1 UPE method is enabled, it will load the script params and the payment methods support features.

Reply:
Sounds great. From what I've tested, things are working as expected, and the issues we saw before aren't happening anymore.

I also think needing this workaround means things could be structured better in the plugin... What do you think? Not thinking of doing it now, though, such a refactor would be big and not worth it at this point, but to start evaluating it for the future.